src/V3EmitCFunc.h

```diff
@@ -148,6 +148,24 @@ class EmitCFunc VL_NOT_FINAL : public EmitCConstInit {
     bool m_useSelfForThis = false;  // Replace "this" with "vlSelf"
     const AstNodeModule* m_modp = nullptr;  // Current module being emitted
     const AstCFunc* m_cfuncp = nullptr;  // Current function being emitted
+    bool m_instantiatesOwnProcess = false;
+
+    bool constructorNeedsProcess(AstClass* classp) {
+        // This function is called only once per emitted constructor code, so this is fine.
+        classp->repairCache();
```

Comment:
Why?  This function doesn't change the class - perhaps this call should move to whatever does.

Reply:
Neither does `TimingSuspendableVisitor::visit(AstCFunc* nodep)`, yet it calls `repairCache`, I assume that's because there is not a single call to it in `V3Task`, despite it modifying the classes heavily.

I still don't know which stage is the culprit here. It's probably not `V3Task`, because `V3Timing` should update the cache after it The lack of this call has caused CI to fail in some system configurations. I wasn't able to reproduce it on my machine. On a different machine it did get reproduced. It all pointed to a faulty lookup and the issue was gone once the call to `repairCache` got added here.

I can investigate it further and make another PR once I find the culprit. That would possibly also remove the `repairCache` call from `V3Timing`.

Reply:
I'd suggest add to the Ast broken() function a test that the cache is correct.  Then every call to V3Broken at each stage will make sure the cache is properly maintained forevermore. Also assuming that assertion then fires, the debugging will be limited to looking at the single visitor source file that ran immediately before the assertion breaks.